### PR TITLE
Mysqlreveng

### PR DESCRIPTION
**Pull request by jespersh**
A few things to note here.
 * https://dev.mysql.com/downloads/windows/visualstudio/ is required.
 * StrongNamer nuget package is used because Pomelo.* aren't strong name signed

### CODE REVIEW

**Comment by jespersh** [comment]
Fixes the mysql part of #17

File: src/GUI/EFCorePowerTools/Resources.Designer.cs [hunk 1]
```diff
@@ -191,6 +191,24 @@ internal class Resources {
             }
         }
         
+        /// <summary>
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Resources.Designer.cs]: Is this needed?

File: src/GUI/EFCorePowerTools/Resources.resx [hunk 2]
```diff
@@ -274,4 +274,10 @@ Storage size varies. It is the length of the value in bytes.</value>
   <data name="NpgsqlProvider" xml:space="preserve">
     <value>70ba90f8-3027-4aF1-9b15-37abbd48744c</value>
   </data>
+  <data name="MysqlDataOdbcProvider" xml:space="preserve">
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Resources.resx]: Is this needed?

File: src/GUI/ReverseEngineer20/ReverseEngineer20.csproj [hunk 3]
```diff
@@ -141,15 +141,24 @@
       <HintPath>..\lib\Microsoft.SqlServer.Types.dll</HintPath>
       <Private>False</Private>
     </Reference>
-    <Reference Include="Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\packages\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
+    <Reference Include="MySqlConnector, Version=0.49.2.0, Culture=neutral, PublicKeyToken=d33d3e53aa5f8c92, processorArchitecture=MSIL">
+      <HintPath>..\packages\MySqlConnector.0.49.2\lib\net471\MySqlConnector.dll</HintPath>
+    </Reference>
+    <Reference Include="Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```
Comment (ErikEJ) [inline src/GUI/ReverseEngineer20/ReverseEngineer20.csproj]: Is this due to MySqlConnector?

File: src/GUI/ReverseEngineer20/packages.config [hunk 4]
```diff
@@ -25,16 +25,20 @@
...
+  <package id="Pomelo.EntityFrameworkCore.MySql" version="2.2.0" targetFramework="net471" />
+  <package id="Pomelo.JsonObject" version="2.2.0" targetFramework="net471" />
   <package id="Remotion.Linq" version="2.2.0" targetFramework="net461" />
   <package id="SQLitePCLRaw.bundle_green" version="1.1.11" targetFramework="net461" />
   <package id="SQLitePCLRaw.core" version="1.1.11" targetFramework="net461" />
   <package id="SQLitePCLRaw.lib.e_sqlite3.linux" version="1.1.11" targetFramework="net461" />
   <package id="SQLitePCLRaw.lib.e_sqlite3.osx" version="1.1.11" targetFramework="net461" />
   <package id="SQLitePCLRaw.lib.e_sqlite3.v110_xp" version="1.1.11" targetFramework="net461" />
   <package id="SQLitePCLRaw.provider.e_sqlite3.net45" version="1.1.11" targetFramework="net461" />
+  <package id="StrongNamer" version="0.0.8" targetFramework="net471" />
```
Comment (ErikEJ) [inline src/GUI/ReverseEngineer20/packages.config]: Does this auto strongname a dll?

File: src/GUI/EFCorePowerTools/Resources.Designer.cs (hunk 1 above)
Comment (jespersh) [reply src/GUI/EFCorePowerTools/Resources.Designer.cs]: Nope. Not the Odbc one. Will remove.

File: src/GUI/EFCorePowerTools/Resources.resx (hunk 2 above)
Comment (jespersh) [reply src/GUI/EFCorePowerTools/Resources.resx]: Not this one

File: src/GUI/ReverseEngineer20/ReverseEngineer20.csproj (hunk 3 above)
Comment (jespersh) [reply src/GUI/ReverseEngineer20/ReverseEngineer20.csproj]: It is from Pomelo.EntityFrameworkCore.MySql depending on  MySqlConnector (>= 0.49.2) and Pomelo.JsonObject which in turn depends on  Newtonsoft.Json (>= 11.0.2)

File: src/GUI/ReverseEngineer20/packages.config (hunk 4 above)
Comment (jespersh) [reply src/GUI/ReverseEngineer20/packages.config]: Yes. The Pomelo.* are auto strongnamed by these. I was surprised it worked.

Comment (ErikEJ) [reply src/GUI/ReverseEngineer20/packages.config]: Amazing!

**Comment by ErikEJ** [comment]
NICE!

**Comment by jespersh** [comment]
Not sure why the online editor on github was inserting the line breaks at the end.
I see you got the changed automatically before I got around to fixing it on a proper dev machine.

**Comment by ErikEJ** [comment]
Donno. Forgot to ask: I assume you have done some smoke testing?

**Comment by jespersh** [comment]
I tested against our product with >500 tables and it only revealed the caveat about the tables only with Unique keys being treated as Primary keys which EFCore couldn't understand. It left those as commented warnings in the Context file.

**Comment by ErikEJ** [comment]
Perfect! Thanks

**Pull request merged**